bundle/filefs.go

```diff
@@ -0,0 +1,69 @@
+// +build go1.16
+
+package bundle
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"sync"
+)
+
+type dirLoaderFS struct {
+	filesystem fs.FS
+	files      []string
+	idx        int
+	mu         sync.Mutex
+}
+
+// NewFSLoader returns a basic DirectoryLoader implementation
+// that will load files from a fs.FS interface
+func NewFSLoader(filesystem fs.FS) DirectoryLoader {
+	d := dirLoaderFS{
+		filesystem: filesystem,
+	}
+
+	return &d
+}
+
+// NextFile iterates to the next file in the directory tree
+// and returns a file Descriptor for the file.
+func (d *dirLoaderFS) NextFile() (*Descriptor, error) {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	// build a list of all files we will iterate over and read, but only one time
+	if d.files == nil {
```

Comment:
@srenatus do you see any reason why this should be in this method and not in `NewFSLoader()`?

Reply:
Not really. I suppose we do it in both places or none at all... There is no concurrent use right now, but if there was, we'd be safe... 🤔 this could go either way...

Reply:
From my point of view, `NewFSLoader()` is only initialized once while `NextFile()` by design is called until no more files. With that in mind, looping through all files and creating the array during initialization would make the code more readable.

Reply:
I have misunderstood your comment at first...I agree on moving this into the other function. 😃

Reply:
@srenatus I'll move it later when I'm back home 😊👍

Reply:
@srenatus I've moved it to the initialization (`NewFSLoader()`) - but it also means it now will return a `DirectoryLoader` and an `error`. WDYT?